pkg/performanceprofile/profilecreator/profilecreator.go

```diff
@@ -289,30 +304,147 @@ func topologyHTDisabled(info *topology.Info) *topology.Info {
+		return cpuset.CPUSet{}, cpuset.CPUSet{}, cpuset.CPUSet{}, fmt.Errorf("can't determine if Hyperthreading is enabled or not: %v", err)
+	}
+
+	// Calculate reserved cpus.
+	reserved, topoInfo, err := ghwHandler.GetReservedCPUs(topologyInfo, reservedCPUCount, splitReservedCPUsAcrossNUMA, disableHTFlag, htEnabled)
+	if err != nil {
+		return cpuset.CPUSet{}, cpuset.CPUSet{}, cpuset.CPUSet{}, err
 	}
+
+	//Calculate offlined cpus
+	// note this takes into account the reserved cpus from the step above
+	offlined, err := ghwHandler.GetOfflinedCPUs(reserved, cpuInfo, topologyInfo, offlinedCPUCount, disableHTFlag, htEnabled)
+	if err != nil {
+		return cpuset.CPUSet{}, cpuset.CPUSet{}, cpuset.CPUSet{}, err
+	}
+
+	// Calculate isolated cpus.
+	// Note that topology info could have been modified by "GetReservedCPUS" so
+	// to properly calculate isolated CPUS we need to use the updated topology information.
+	isolated, err := ghwHandler.GetIsolatedCPUs(topoInfo.Nodes, reserved, offlined)
+	if err != nil {
+		return cpuset.CPUSet{}, cpuset.CPUSet{}, cpuset.CPUSet{}, err
+	}
+
+	return reserved, isolated, offlined, nil
+}
+
+// Calculates Isolated cpuSet as the difference between all the cpus in the topology and those already choosen as reserved or offlined.
+// all cpus thar are not offlined or reserved belongs to the isolated cpuSet
+func (ghwHandler GHWHandler) GetIsolatedCPUs(topologyInfoNodes []*topology.Node, reserved, offlined cpuset.CPUSet) (cpuset.CPUSet, error) {
+	total, err := totalCPUSetFromTopology(topologyInfoNodes)
+	if err != nil {
+		return cpuset.CPUSet{}, err
+	}
+	return total.Difference(reserved.Union(offlined)), nil
+}
+
+func (ghwHandler GHWHandler) GetOfflinedCPUs(alreadyUsedCPUs cpuset.CPUSet, cpuInfo *cpu.Info, topologyInfo *topology.Info, offlinedCPUCount int, disableHTFlag bool, htEnabled bool) (cpuset.CPUSet, error) {
+	offlinedCPUs := newCPUAccumulator()
+	lpOfflined := 0
+
+	//currently HT is enabled on the system and the user wants to disable HT
+	if htEnabled && disableHTFlag {
+		for _, node := range topologyInfo.Nodes {
+			acc, err := offlinedCPUs.AddCoresWithFilter(offlinedCPUCount, node.Cores, func(index, lpID int) bool {
+				return !alreadyUsedCPUs.Contains(lpID) && filterFirstLogicalProcessorInCore(index, lpID)
+			})
+			if err != nil {
+				return cpuset.CPUSet{}, err
+			}
+			lpOfflined += acc
+		}
+		if lpOfflined < offlinedCPUCount {
+			for _, node := range topologyInfo.Nodes {
+				acc, err := offlinedCPUs.AddCoresWithFilter(offlinedCPUCount, node.Cores, func(index, lpID int) bool {
+					return !alreadyUsedCPUs.Contains(lpID)
+				})
+				if err != nil {
+					return cpuset.CPUSet{}, err
+				}
+				lpOfflined += acc
+			}
+		}
+	} else {
+		for _, processor := range cpuInfo.Processors {
+			if processor.NumThreads <= uint32(offlinedCPUCount-lpOfflined) {
+				// We can offline a complete CPU socket
```

Comment:
same, do we have a e2e test which ensures that we picked all the cpus pertaining a socket? e.g. nothing more? (and probably nothing less?)

Reply:
Yes and no .... 
There is a test addressing this case but to disable a complete socket is almost impossible right now. 
As reservedCPUs are calculated first and we do not have socket layout into account to do it, it is almost certain that at least one cpu of each socket would be reserved.

Reply:
anyway, we can hunt and use more system topologies if this helps us

Reply:
> But we can reserve only cpus from socket 0 and offline all socket 1 in theory, can't we?

Surely we could do it, but right now we are not doing it at all. We do not consult CPU info to choose reserved CPUs, just TopologyInfo so we just choose the cpus in the same NUMA zone or balance them between the NUMA zones but we do not really know if they belong to the same or different sockets.

Reply:
ok, so we need to first clarify/unify our own terminology and then iterate from there. Even with multi-NUMA single-socket CPUs offlining an entire NUMA zone can yield significant improvements.